tests/ga/test_extension.py

```diff
@@ -1241,6 +1252,122 @@ def test_upgrade_failure(self, patch_get_update_command, *args):
 
         self._assert_handler_status(protocol.report_vm_status, "NotReady", expected_ext_count=1, version="1.0.1")
 
+    @patch('azurelinuxagent.ga.exthandlers.HandlerManifest.get_disable_command')
+    def test__exgension_upgrade_failure_when_prev_version_disable_fails(self, patch_get_disable_command, *args):
+        test_data = WireProtocolData(DATA_FILE_EXT_SINGLE)
+        exthandlers_handler, protocol = self._create_mock(test_data, *args)
+
+        # Ensure initial install and enable is successful
+        exthandlers_handler.run()
+        self.assertEqual(0, patch_get_disable_command.call_count)
+
+        self._assert_handler_status(protocol.report_vm_status, "Ready", expected_ext_count=1, version="1.0.0")
+        self._assert_ext_status(protocol.report_ext_status, "success", 0)
+
+        # Next incarnation, update version
+        test_data.goal_state = test_data.goal_state.replace("<Incarnation>1<",
+                                                            "<Incarnation>2<")
+        test_data.ext_conf = test_data.ext_conf.replace('version="1.0.0"',
+                                                        'version="1.0.1"')
+        test_data.manifest = test_data.manifest.replace('1.0.0',
+                                                        '1.0.1')
+
+        # Disable of the old extn fails
+        patch_get_disable_command.return_value = "exit 1"
+        exthandlers_handler.run()  # Download the new update the first time, and then we patch the download method.
+
+        with patch('azurelinuxagent.common.protocol.restapi.Protocol.download_ext_handler_pkg') as patch_download:
+            loop_run = 5
+            for x in range(loop_run):
+                exthandlers_handler.run()
+
+            self.assertEqual(loop_run + 1, patch_get_disable_command.call_count)  # counting the earlier run done to
+            # donwload the new update
+            self.assertEqual(0, patch_download.call_count)  # The download should never be called.
+
+        # On the next iteration, update should not be retried
+        self._assert_handler_status(protocol.report_vm_status, "NotReady", expected_ext_count=0, version="1.0.1")
+
+    @patch('azurelinuxagent.ga.exthandlers.HandlerManifest.get_disable_command')
+    def test__exgension_upgrade_failure_when_prev_version_disable_fails_and_recovers(self, patch_get_disable_command, *args):
```

Comment:
nit: same as above

Reply:
Isn't this test the same as the one above?

Reply:
They are not the same:
```
jking@dvm:~/WALinuxAgent$ diff <(sed -n "1255,1289p" tests/ga/test_extension.py) <(sed -n "1291,1325p" tests/ga/test_extension.py)
2c2
<     def test__exgension_upgrade_failure_when_prev_version_disable_fails(self, patch_get_disable_command, *args):
---
>     def test__exgension_upgrade_failure_when_prev_version_disable_fails_and_recovers(self, patch_get_disable_command, *args):
```

Reply:
Fixed. Yes they were same - it was a mistake from my end, the file wasn't updated.